src/gp/kernel.jl

```diff
@@ -201,6 +201,25 @@ pw(k::RQ, x::AV) = _rq.(pw(SqEuclidean(), x), k.α)
 
 
 
+"""
+    Cosine <: Kernel
+
+Cosine Kernel with period parameter `p`.
+"""
+struct Cosine{Tp<:Real} <: Kernel
+    p::Tp
+end
+
+# Binary methods.
+ew(k::Cosine, x::AV, x′::AV) = cos.(pi.*ew(Euclidean(), x, x′) ./k.p)
+pw(k::Cosine, x::AV, x′::AV) = cos.(pi.*pw(Euclidean(), x, x′) ./k.p)
+
+# Unary methods.
+ew(k::Cosine, x::AV) = ones(eltype(x), length(x))
```

Comment:
Looks like tests are still failing when `eltype(x)` is a `Vector` i.e. for vector-valued inputs. Maybe use a similar pattern to the `Matern` kernels and just utilise the fact that `ew(Euclidean(). x)` always returns a vector of zeros. It's a little inefficient, but ultimately not too bad.

Reply:
since we need the output to be one 
I have modified it to ```1 .+ ew(Euclidean(), x)```